src/swap/components/SwapButton.tsx

```diff
@@ -7,6 +7,7 @@ export function SwapButton({
   className,
   disabled = false,
   onError,
+  onStart,
```

Comment:
We are moving more and more into the land of creating a component life-cycle.

Those are big decision to take, so let's take a minute to reflect on every step we are taking.

Do we see `onStart` being used in other components as well? And overall what kind of expecation we want to train the developer when they see `onStart` into a component?

Reply:
this is a specific customer request from PartyDAO to manage the component life-cycle, so they can implement their own toasts

regardless, even if we ship TransactionKit, we should expose hooks that allow the developer to customize the transaction states (in this instance, removing our `SwapMessage` and using their own)

Reply:
since we already have `onSuccess` (which imo is pretty intuitive since the developer may want to make a callback with the resulting swap transaction hash), I think the customer request for an `onStart` is reasonable as well

this specific use case they want to add their own toast, but I can imagine it being used for other reasons as well

Reply:
> I'm of the opinion we should expose hooks that allow the developer to customize the transaction states

I fully agree on this, in fact if you remember I mentioned a few times we should have a `onStatus` hook, to give that level of customization.

Do we think `onStatus` can satisfay `onStart` and future lifecycle states? What I want to avoid is creating 3 more custom listeners, but instead have one that listens all states and get give great inisights on what's going on.

Ideally `onStatus` could deprecate `onSuccess` at some point.

Reply:
added `onStatus` prop